state/azure/cosmosdb/cosmosdb.go

```diff
@@ -208,3 +268,70 @@ func (c *StateStore) BulkDelete(req []state.DeleteRequest) error {
 
 	return nil
 }
+
+// Multi performs a transactional operation. succeeds only if all operations succeed, and fails if one or more operations fail
+func (c *StateStore) Multi(operations []state.TransactionalRequest) error {
+	upserts := []CosmosItem{}
+	deletes := []CosmosItem{}
+
+	partitionKey := unknownPartitionKey
+	previousPartitionKey := unknownPartitionKey
+
+	for _, o := range operations {
+		if o.Operation == state.Upsert {
+			req := o.Request.(state.SetRequest)
+
+			partitionKey = populatePartitionMetadata(req.Key, req.Metadata)
```

Comment:
Lines 284 - 288 and 300 - 304 are essentially the same and can move up to line 281.